### ISSUE TITLE
Broken thumb handling when jobs file still exists

### ISSUE THREAD

**Issue by nilshoerrmann** [k-next/kirby#4632]
## Description
We ran into the situation where thumb generation for a large image failed (converting from JPEG source to WebP thumb): the generated thumb was only half rendered, the lower part missing, so I guess there was some kind of timeout while writing to disk. Kirby did not delete the corresponding jobs file so the media folder was in a conflicting state: we had both, a generated (but broken) thumb and a jobs file. 

When requesting the image, Kirby served the broken thumb.

**Expected behavior**  
When Kirby detects an existing jobs file, it should always generate a new thumb because most certainly something failed when creating any existing thumb.

## Your setup
**Kirby Version**  
3.7.5

**Comment by nilshoerrmann** [comment]
Maybe related: #4249

**Comment by lukasbestle** [comment]
## Status quo

Kirby first generates the thumb in its destination and then deletes the job file. If anything goes wrong (= if any exception is thrown during the thumb generation), the job file is kept, the broken thumb file is deleted and the source is returned (see discussion in #4249).

https://github.com/getkirby/kirby/blob/c7f52c4932d637c7eda090095285a73119a24456/src/Cms/Media.php#L126-L133

## Technical background

With the default `.htaccess` (and recommended server setup), requests to the `media` folder are first served by the web server. Only if the thumb file does not exist, Kirby is spun up to generate the thumb.

So if there is ever a broken thumb file in the media folder, Kirby can never act on this because the web server will never execute Kirby. This means that we cannot check if there is a job file.

## Why the broken thumb file might have been written

The question is: Why is there a broken thumb file in the first place? If there had been an exception, the thumb file would have been deleted. But maybe the PHP process got killed unexpectedly. This is something we can never catch directly.

## Possible solution

We can generate the thumb in a temporary location and only move it to the destination on success. This would be pretty simple to do with PHP `tmpfile()`.

**Comment by distantnative** [comment]
@lukasbestle you write
>  If there had been an exception, the thumb file would have been deleted. But maybe the PHP process got killed unexpectedly.

If that's the problem - wouldn't that likely still be the problem if we write it to a tmp location and try to move it?

**Comment by lukasbestle** [comment]
If the process is killed right when the file is moved, yes. But I'd say that it's a lot more likely that the process is killed during thumb generation (which is a way more resource-intensive and time-consuming step).

**Comment by mrflix** [comment]
Good possible solution, @lukasbestle!

**Comment by nilshoerrmann** [comment]
I like the proposed solution :+1:

**Comment by distantnative** [comment]
Trying to revive this discussion and make it more concrete:


Currently, in the `thumb` core component we do 

```php
F::copy($src, $root, true);
$darkroom->process($root, $options);
```

based on the proposed solution, we would change this to something like

```php
$tmp = tempnam(sys_get_temp_dir(), 'TMP_');
F::copy($src, $tmp, true);
$darkroom->process($root, $options);
F::copy($tmp, $root, true);
```

?

That way, if something in the process fails that doesn't trigger an exception, the job file is not removed, but also no media file is present in the final location. So with the next request, Kirby would try again to process the job file.

**Comment by lukasbestle** [comment]
Yes, pretty much. Except I'd use `tmpfile()` so that PHP ensures that the temporary file is cleaned up afterwards (even on most errors):

```php
$tmp = tmpfile();
$tmpPath = stream_get_meta_data($tmp)['uri'];

F::copy($src, $tmpPath, true);
$darkroom->process($tmpPath, $options);
F::copy($tmpPath, $root, true);
```

**Comment by afbora** [comment]
I think this issue can be addressed more reliably by making the default thumb generation flow safer instead of trying to clean up an already-written thumb in `Media::thumb()`.

Right now, the default `thumb` component writes directly to the final destination path. That means if generation is interrupted halfway through, or if the process exits unexpectedly, a partially written or otherwise broken thumb can still be left behind under its final filename. In that situation, the web server may serve that physical file directly without the request ever reaching Kirby again, so checks in `Media::thumb()` won’t always get a chance to correct the state.

Because of that, a safer approach would be to generate the thumb in a temporary file in the same directory first, and only move it to the final destination once processing has completed successfully. 

This way:
- if generation fails, no broken file ever appears at the final thumb path
- if generation succeeds, the final file appears in one step
- unexpected interruptions are much less likely to leave behind a broken thumb that can be served directly

So the suggested flow would be:
1. Copy the source to a temporary thumb file
2. Run the darkroom processing on the temporary file
3. If processing succeeds, move the temporary file to the final destination
4. If processing fails, remove the temporary file and rethrow the exception

I think this keeps the fix fairly small, stays aligned with the existing architecture, and is more robust than trying to delete the final output in a `catch` block after it may already have become visible at the final path.

What do you think about the solution? @getkirby/backend

**Comment by bastianallgeier** [comment]
@afbora I really like that idea. Could still be in the same folder but with a differnt name or extension though, right?

**Comment by afbora** [comment]
@bastianallgeier Exactly. In the same folder, with the same extension, but with a different and unique filename.

Here's the small change I'm suggesting.

<img width="768" height="521" alt="Image" src="https://github.com/user-attachments/assets/11681bdf-bbc3-4c00-9d8c-1a0ee85acf60" />

**Comment by bastianallgeier** [comment]
Looks good on first sight. Could you create a PR for it?

**Comment by silllli** [comment]
Thank you, that's a great improvement!
